packages/block-library/src/navigation/index.php

```diff
@@ -254,7 +254,7 @@ function render_block_core_navigation( $attributes, $content, $block ) {
 	$inner_blocks = $block->inner_blocks;
 
 	// If `__unstableLocation` is defined, create inner blocks from the classic menu assigned to that location.
-	if ( empty( $inner_blocks ) && array_key_exists( '__unstableLocation', $attributes ) ) {
+	if ( ! array_key_exists( 'navigationMenuId', $attributes ) && array_key_exists( '__unstableLocation', $attributes ) ) {
```

Comment:
```suggestion
	if ( empty( $inner_blocks ) || ! array_key_exists( 'navigationMenuId', $attributes ) && array_key_exists( '__unstableLocation', $attributes ) ) {
```

Should we still include the `empty( $inner_blocks )` check for backwards compat with the older way of storing nav block "items" as inner blocks?

Reply:
Wait, was it ever deprecated? I thought this a fully supported feature that powers navigation in block patterns.

Reply:
How about `&&` instead of `||`?

	if ( empty( $inner_blocks ) && ! array_key_exists( 'navigationMenuId', $attributes ) && array_key_exists( '__unstableLocation', $attributes ) ) {

Reply:
> Wait, was it ever deprecated?

Oh right. I'm so confused. So the block can have inner blocks as long as they aren't added via the UI. Is that it?

Reply:
I think so, see https://github.com/WordPress/gutenberg/pull/36024 and https://github.com/WordPress/gutenberg/issues/35947

> This PR automatically creates an untitled wp_navigation menu whenever:
> A pattern with uncontrolled/unsaved inner blocks is inserted
> A post/template has a navigation block with unsaved block in it already, and the editor is loaded.

Reply:
Answer yes. I updated the code.

Reply:
Yes that's right. We added support to allow things like block patterns to still add inner blocks. Therefore a Nav block can be considered "populated" either if:

- it has inner blocks
- it has a pointer to a `wp_navigation` CPT post ID.